src/rust/catmem/mod.rs

```diff
@@ -296,3 +300,23 @@ impl CatmemLibOS {
         self.scheduler.poll()
     }
 }
+
+//==============================================================================
+// Standalone Functions
+//==============================================================================
+
+// Pushes EoF.
+fn push_eof(ring: Rc<SharedRingBuffer<u16>>) -> Result<(), Fail> {
+    let x: u16 = ((1 & 0xff) << 8) as u16;
+
+    loop {
+        match ring.try_enqueue(x) {
+            Ok(()) => break,
+            Err(_) => {
+                warn!("failed to push EoF")
+            },
+        }
+    }
+
+    Ok(())
+}
```

Comment:
I need to call it from an async function that does not have a reference to self. Since there is no need to reference self in this function, I made it standalone.

Reply:
Still you could have this in the scope of the structure. Just leave it there and remove the self parameter.